core/domain/rights_manager.py

```diff
@@ -184,18 +214,45 @@ def _save_activity_rights(
 # same as the activity id).
 # TODO(msl): get rid of inline imports by refactoring code
 def _update_exploration_summary(activity_rights):
+    """Updates the exploration summary field for an activity.
+
+    Args:
+        activity_rights: ActivityRights. Rights object for this activity.
+
+    Returns:
```

Comment:
When there is no Return value, you don't need to include these two lines (and the empty line above). Ditto throughout the rest of this PR.